### PR TITLE
Mail contacts

### PR DESCRIPTION
**Pull request by Derekljh**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/MailCommand.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        for (int i = 0; i < model.getFilteredPersonList().size(); i++) {
+            if (i == model.getFilteredPersonList().size() - 1) {
+                builder.append(model.getFilteredPersonList().get(i).getEmail());
+            } else {
+                builder.append(model.getFilteredPersonList().get(i).getEmail()).append(",");
+            }
+        }
+        try {
+            Desktop desktop;
+            if (Desktop.isDesktopSupported()
```
Comment (radeon2525) [inline src/main/java/seedu/address/logic/commands/MailCommand.java:54]: Could you tell what .isDesktopSupported() does, is it checking for compatibility with the OS?

File: src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.StringUtil;
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code Name} matches any of the keywords given.
+ */
+public class TagContainsKeywordsPredicate implements Predicate<Person> {
```
Comment (radeon2525) [inline src/main/java/seedu/address/model/person/TagContainsKeywordsPredicate.java:12]: LGTM

File: src/test/java/seedu/address/model/person/TagContainsKeywordsPredicateTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.testutil.PersonBuilder;
+
+public class TagContainsKeywordsPredicateTest {
```
Comment (radeon2525) [inline src/test/java/seedu/address/model/person/TagContainsKeywordsPredicateTest.java:15]: LGTM

File: src/test/java/seedu/address/logic/parser/MailCommandParserTest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.MailCommand;
+import seedu.address.model.person.TagContainsKeywordsPredicate;
+
+public class MailCommandParserTest {
+
+    private MailCommandParser parser = new MailCommandParser();
+
+    @Test
```
Comment (radeon2525) [inline src/test/java/seedu/address/logic/parser/MailCommandParserTest.java:18]: LGTM

File: src/test/java/seedu/address/logic/commands/MailCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import seedu.address.logic.CommandHistory;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.TagContainsKeywordsPredicate;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code FindCommand}.
+ */
+public class MailCommandTest {
```
Comment (radeon2525) [inline src/test/java/seedu/address/logic/commands/MailCommandTest.java:23]: Maybe can try adding a few more tests as this is the main class addition for this PR to increase the codecov coverage

Comment (Derekljh) [reply src/test/java/seedu/address/logic/commands/MailCommandTest.java:23]: i did try to add the tests but it passes on my end but not on the CI end

File: src/main/java/seedu/address/logic/commands/MailCommand.java (hunk 1 above)
Comment (Derekljh) [reply src/main/java/seedu/address/logic/commands/MailCommand.java:54]: to determine if the current desktop is supported. Tests whether an action is supported on the current platform.

**Pull request merged**